### PR TITLE
command: add `sub-ass-extradata` property

### PR DESCRIPTION
**Pull request by llyyr**
Added so I can do what #12157 does through a lua script (with sub-ass-force-style) instead, since it seems there isn't much interest in that PR.

I still prefer that PR for this but it seems mpv don't want too many hacks around libass

### CODE REVIEW

File: player/command.c [hunk 1]
```diff
@@ -2916,6 +2916,29 @@ static int mp_property_sub_pos(void *ctx, struct m_property *prop,
     return mp_property_generic_option(mpctx, prop, action, arg);
 }
 
+static int mp_property_sub_extradata(void *ctx, struct m_property *prop,
+                                     int action, void *arg)
+{
+    MPContext *mpctx = ctx;
+    struct track *track = mpctx->current_track[0][STREAM_SUB];
```
Comment (Dudemanguy) [inline player/command.c:2923]: `current_track` can be null

Comment (llyyr) [reply player/command.c:2923]: Isn't the next line sufficient? I checked other functions and that's how we do it everywhere.

```
    struct dec_sub *sub = track ? track->d_sub : NULL;
    if (!sub)
        return M_PROPERTY_UNAVAILABLE;
```

I also made sure to test with files with no subtitles, or files with subs but none selected and it returns property unavailable without exploding like it should.

Comment (Dudemanguy) [reply player/command.c:2923]: Ah you're right. I misunderstood the comment in `core.h`. The `current_track[MAX_PTRACKS][STREAM_TYPE_COUNT]` is what's NULL not `current_track`.

File: DOCS/man/input.rst [hunk 2]
```diff
@@ -2679,6 +2679,10 @@ Property list
         coordinates should be ignored when this value is false, because the
         video backends update them only when the pointer hovers the window.
 
+``sub-extradata``
+    The current subtitle track's extradata. There is no formatting done, the
```
Comment (Dudemanguy) [inline DOCS/man/input.rst:2683]:
```suggestion
    The current subtitle track's extradata. There is no formatting done. The
```
Small nit.

File: sub/dec_sub.c [hunk 3]
```diff
@@ -369,6 +369,11 @@ char *sub_get_text(struct dec_sub *sub, double pts, enum sd_text_type type)
     return text;
 }
 
+char *sub_get_extradata(struct dec_sub *sub)
+{
+    return talloc_strdup(NULL, sub->sd->codec->extradata);
```
Comment (Dudemanguy) [inline sub/dec_sub.c:374]: This actually causes a buffer overflow on one of my files (a subplease rip). I think you might need to use `talloc_strndup` with `extradata_size` as the size of the buffer.

File: sub/dec_sub.c [hunk 4]
```diff
@@ -369,6 +369,15 @@ char *sub_get_text(struct dec_sub *sub, double pts, enum sd_text_type type)
     return text;
 }
 
+char *sub_ass_get_extradata(struct dec_sub *sub)
+{
+    if (bstr_find0(bstr0(sub->sd->codec->codec), "ass") < 0)
```
Comment (Dudemanguy) [inline sub/dec_sub.c:374]: Why not just `strcmp(sub->sd->codec->codec, "ass")`?

Comment (llyyr) [reply sub/dec_sub.c]: For some reason I thought it wouldn't be that simple without even trying... just doing strcmp works though, changed it.

**Review by Dudemanguy — APPROVED** [review]

File: sub/dec_sub.c [hunk 5]
```diff
@@ -369,6 +369,15 @@ char *sub_get_text(struct dec_sub *sub, double pts, enum sd_text_type type)
     return text;
 }
 
+char *sub_ass_get_extradata(struct dec_sub *sub)
+{
+    if (strcmp(sub->sd->codec->codec, "ass") < 0)
```
Comment (Dudemanguy) [inline sub/dec_sub.c:374]:
```suggestion
    if (strcmp(sub->sd->codec->codec, "ass") != 0)
```
Oops, missed that yesterday apparently.
